engine/src/main/java/io/zeebe/engine/processor/workflow/handlers/container/ProcessCompletedHandler.java

```diff
@@ -38,8 +44,49 @@ protected boolean handleState(BpmnStepContext<ExecutableFlowElementContainer> co
                 WorkflowInstanceIntent.ELEMENT_COMPLETING,
                 parentElementInstance.getValue());
       }
+    } else {
+      sendProcessCompletedResult(context);
     }
 
     return super.handleState(context);
   }
+
+  private void sendProcessCompletedResult(BpmnStepContext context) {
+
+    final long elementInstanceKey = context.getElementInstance().getKey();
+    final AwaitWorkflowInstanceResultMetadata requestMetadata =
+        context.getElementInstanceState().getAwaitResultRequestMetadata(elementInstanceKey);
+    if (requestMetadata != null) {
+      final DirectBuffer variablesAsDocument =
+          context
+              .getElementInstanceState()
+              .getVariablesState()
+              .getVariablesAsDocument(elementInstanceKey);
+
+      final WorkflowInstanceResultRecord resultRecord = new WorkflowInstanceResultRecord();
+      resultRecord
+          .setWorkflowInstanceKey(context.getValue().getWorkflowInstanceKey())
+          .setWorkflowKey(context.getValue().getWorkflowKey())
+          .setVariables(variablesAsDocument)
+          .setBpmnProcessId(context.getValue().getBpmnProcessId())
+          .setVersion(context.getValue().getVersion());
+
+      context
+          .getOutput()
+          .getStreamWriter()
+          .appendFollowUpEvent(
```

Comment:
Do we want this on the log stream? I mean there was a reason why we removed the variables from the records that is why I'm asking.

Reply:
It is not needed for correctness. It was added for observability. It is also nice to have it for testing. But it can be removed. What was the reason for removing variables from the records?